Wikipedia/Code/WMFThemeableNavigationController.m

```diff
@@ -65,53 +65,37 @@ - (void)applyTheme:(WMFTheme *)theme {
 
 #pragma mark - Splash
 
-- (UIImageView *)splashView {
-    if (!_splashView) {
-        _splashView = [[UIImageView alloc] init];
-        _splashView.contentMode = UIViewContentModeCenter;
-        if (UI_USER_INTERFACE_IDIOM() != UIUserInterfaceIdiomPad) {
-            [_splashView setImage:[UIImage imageNamed:@"splashscreen-background"]];
-        }
-        if (@available(iOS 13.0, *)) {
-            _splashView.backgroundColor = [UIColor systemBackgroundColor];
-        } else {
-            _splashView.backgroundColor = [UIColor whiteColor];
-        }
-        [self.view wmf_addSubviewWithConstraintsToEdges:_splashView];
-        UIImage *wordmark = [UIImage imageNamed:@"splashscreen-wordmark"];
-        UIImageView *wordmarkView = [[UIImageView alloc] initWithImage:wordmark];
-        wordmarkView.translatesAutoresizingMaskIntoConstraints = NO;
-        [_splashView addSubview:wordmarkView];
-        NSLayoutConstraint *centerXConstraint = [_splashView.centerXAnchor constraintEqualToAnchor:wordmarkView.centerXAnchor];
-        NSLayoutConstraint *centerYConstraint = [_splashView.centerYAnchor constraintEqualToAnchor:wordmarkView.centerYAnchor constant:12];
-        [_splashView addConstraints:@[centerXConstraint, centerYConstraint]];
-    }
-    return _splashView;
-}
-
 - (void)showSplashView {
-    self.splashView.hidden = NO;
-    self.splashView.alpha = 1.0;
-}
-
-- (void)showSplashViewIfNotShowing {
-    if (!self.splashView.hidden) {
+    if (self.splashScreenViewController) {
         return;
     }
-    [self showSplashView];
+    WMFSplashScreenViewController *splashVC = [[WMFSplashScreenViewController alloc] init];
+    // Explicit appearance transitions need to be used here because UINavigationController overrides
+    // a lot of behaviors when adding the VC as a child and causes layout issues for our use case.
```

Comment:
Hmm not really sold on this, feels like a sign that we shouldn't bother if nav view controller doesn't play nicely with a child view controller. It also feels weird for all WMFThemeableNavigationControllers to have the ability to show/hide a splash view if only one instance is ever going to use it. I know AppVC is huge but still seems like something that only it should have the logic to do. I'm open to leaving as-is since that's existing but wanted to say my two cents.

Reply:
>Hmm not really sold on this, feels like a sign that we shouldn't bother if nav view controller doesn't play nicely with a child view controller. 

It seems the built-in parent/child VC methods are more for convenience rather than a requirement. Given these appearance transition methods aren't deprecated, I feel OK about using them.

>  It also feels weird for all WMFThemeableNavigationControllers to have the ability to show/hide a splash view if only one instance is ever going to use it. 

Would it help to move this functionality to a separate `RootNavigationController` class?

> I know AppVC is huge but still seems like something that only it should have the logic to do. I'm open to leaving as-is since that's existing but wanted to say my two cents.

Unfortunately AppVC is poorly named as it isn't the root VC of the app. It's the root VC of a `WMFThemeableNavigationController` that is the root VC of the app. As a result, the appVC would have to present the splash screen modally off of its navigation controller anyway to cover any pushed VCs. I tried this approach, but the transition wasn't smooth, it didn't seem like we could control the cross fade duration, and it potentially conflicts with the modal presentation of onboarding.

Reply:
> Would it help to move this functionality to a separate RootNavigationController class?

That sounds good, I think that would cover it all for me. And good point on those methods not being deprecated so should be fine. On your modal point, I can see how that would mess things up. I was initially more thinking have the view still live on the navigationController's view, but AppVC owns the logic that adds the splash view, swaps it out with the loading animation, etc. But since loading animation is already in a view controller it'd be more rework and now that I'm typing out-loud feels like moving backwards to cram more stuff into AppVC.

Reply:
I moved the functionality into a separate `RootNavigationController`. Long term it'd be good to explore turning that into a plain view controller that has the `WMFThemeableNavigationController` and the `SplashScreenViewController` as a child view controllers, but for now given we're late in this release it makes sense to keep it as close to the old VC structure as we can.